tests/phpunit/tests/dependencies.php

```diff
@@ -148,4 +148,94 @@ public function test_enqueue_before_register() {
 
 		$this->assertContains( 'one', $dep->queue );
 	}
+
+	/**
+	 * Data provider for test_get_etag.
+	 *
+	 * @return array[]
+	 */
+	public function data_provider_get_etag() {
+		return array(
+			'should accept one dependency'              => array(
+				'load'               => array(
+					'abcd' => '1.0.2',
+				),
+				'hash_source_string' => 'WP:6.7;abcd:1.0.2;',
+				'expected'           => 'W/"8145d7e3c41d5a9cc2bccba4afa861fc"',
+			),
+			'should accept empty array of dependencies' => array(
+				'load'               => array(),
+				'hash_source_string' => 'WP:6.7;',
+				'expected'           => 'W/"7ee896c19250a3d174f11469a4ad0b1e"',
+			),
+		);
+	}
+
+	/**
+	 * Tests get_etag method for WP_Scripts.
+	 *
+	 * @ticket 58433
+	 * @ticket 61485
+	 *
+	 * @covers WP_Dependencies::get_etag
+	 *
+	 * @dataProvider data_provider_get_etag
+	 *
+	 * @param array $load                List of scripts to load.
+	 * @param string $hash_source_string Hash source string.
+	 * @param string $expected           Expected etag.
+	 */
+	public function test_get_etag_scripts( $load, $hash_source_string, $expected ) {
+		global $wp_version;
+		// Modify global to avoid tests needing to change with each new version of WordPress.
+		$original_wp_version = $wp_version;
+		$wp_version          = '6.7';
+		$instance            = wp_scripts();
+
+		foreach ( $load as $handle => $ver ) {
+			// The src should not be empty.
+			wp_enqueue_script( $handle, 'https://example.org', array(), $ver );
+		}
+
+		$result = $instance->get_etag( array_keys( $load ) );
+
+		// Restore global prior to making assertions.
+		$wp_version = $original_wp_version;
+
+		$this->assertSame( $expected, $result, "Expected MD hash: $expected for $hash_source_string, but got: $result." );
+	}
+
+	/**
+	 * Tests get_etag method for WP_Styles.
+	 *
+	 * @ticket 58433
+	 * @ticket 61485
+	 *
+	 * @covers WP_Dependencies::get_etag
+	 *
+	 * @dataProvider data_provider_get_etag
+	 *
+	 * @param array $load                List of scripts to load.
+	 * @param string $hash_source_string Hash source string.
+	 * @param string $expected           Expected etag.
```

Comment:
Alignment is also probably my fault here too.

```suggestion
	 * @param array  $load               List of styles to load.
	 * @param string $hash_source_string Hash source string.
	 * @param string $expected           Expected etag.
```